text/0000-repr-type-aliases.md

```diff
@@ -20,25 +20,25 @@ For the same reasons why type aliases are useful, having type aliases in `repr`
 # Guide-level explanation
 [guide-level-explanation]: #guide-level-explanation
 
-Enums allow `#[repr(type)]` attributes to offer an explicit discriminant type. (`type` can be any primitive integer type, like `u8`, `i32`, or `usize`, but not `char`.) If all variants of the enum are unit variants, this means that the enum will be easily castable to `type` using `as`. Otherwise, the discriminant will still be of the specified type, but unsafe code is required to actually access it.
+Enums allow `#[repr(type = ...)]` attributes to offer an explicit discriminant type. (`...` can be any primitive integer type, like `u8`, `i32`, or `usize`, but not `char`.) If all variants of the enum are unit variants, this means that the enum will be easily castable to `type` using `as`. Otherwise, the discriminant will still be of the specified type, but unsafe code is required to actually access it.
 
-In addition to the primitive types themselves, you can also use the path to a type alias in the `repr` attribute instead, and it will resolve the primitive type of the type alias. However, to ensure compatibility as new potential representations are added, the path to the alias must contain a double-colon: you can access an alias `Alias` defined in the same module by using `self::Alias`.
-
-For example, `#[repr(core::ffi::c_int)]` is valid because it contains a double-colon, but a `use core::ffi::c_int` followed by `#[repr(c_int)]` is not. If you wanted to `use core::ffi::c_int` first, then you could still do `#[repr(self::c_int)]` to reference the type.
+To ensure compatibility, the `#[repr(type = ...)]` form is required if the type is not one of the known primitive types. Note that this form is not necessarily equivalent to using the primitive representations directly, since shadowing is possible; for example, if you did `type u32 = u8` and then `#[repr(type = u32)]`, this would be equivalent to `#[repr(u8)]`, not `#[repr(u32)]`.
```

Comment:
I think `type u32 = u8` seems needlessly obfuscating; I think it'd be a more readable example to write `type C = u8` and then `#[repr(type = C)]`, which is equivalent to `#[repr(u8)]` rather than `$[repr(C)]`.

Reply:
I guess that I was intentionally pointing out the obfuscation here because it feels more likely: `#[repr(type = C)]` is obviously wrong, but `#[repr(type = u32)]` meaning `#[repr(u8)]` is more likely to occur in something like proc macros if someone is doing something nefarious. So, genuinely, there is a preference to do `#[repr(u32)]` over `#[repr(type = u32)]` when you don't necessarily trust the parent scope.

Reply:
I've since updated this section a bit to elaborate a bit better, including what a type alias `C` might look like. Does this feel like it addresses your concerns?